File: test_samples/sample.py
# Python test sample with various constructs
import asyncio
from typing import List, Dict
import json

class DataProcessor:
    """A sample data processing class."""
    
    def __init__(self, name: str):
        self.name = name
        self.data = []
    
    @staticmethod
    def create_default():
        return DataProcessor("default")
    
    async def process_data(self, items: List[Dict]) -> Dict:
        """Process data asynchronously."""
        results = {}
        for item in items:
            processed = await self.process_item(item)
            results[item['id']] = processed
        return results
    
    async def process_item(self, item: Dict) -> str:
        """Process a single item."""
        await asyncio.sleep(0.01)  # Simulate async work
        return f"processed_{item.get('name', 'unknown')}"
    
    def get_stats(self) -> Dict[str, int]:
        return {
            'total_items': len(self.data),
            'name_length': len(self.name)
        }

# Top-level functions
def main():
    processor = DataProcessor("test")
    data = [{'id': 1, 'name': 'item1'}, {'id': 2, 'name': 'item2'}]
    
    async def run():
        result = await processor.process_data(data)
        print(json.dumps(result, indent=2))
    
    asyncio.run(run())

if __name__ == "__main__":
    main()

File: test_samples/sample.cpp
// C++ test sample with various constructs
#include <iostream>
#include <vector>
#include <memory>
#include <string>

class DataProcessor {
private:
    std::string name;
    std::vector<int> data;

public:
    DataProcessor(const std::string& name) : name(name) {}
    
    virtual ~DataProcessor() = default;
    
    static std::unique_ptr<DataProcessor> create_default() {
        return std::make_unique<DataProcessor>("default");
    }
    
    virtual void process_data(const std::vector<int>& items) {
        for (const auto& item : items) {
            process_item(item);
        }
    }
    
    inline void process_item(int item) {
        data.push_back(item * 2);
    }
    
    const std::vector<int>& get_data() const {
        return data;
    }
};

class AdvancedProcessor : public DataProcessor {
public:
    AdvancedProcessor() : DataProcessor("advanced") {}
    
    void process_data(const std::vector<int>& items) override {
        std::cout << "Advanced processing..." << std::endl;
        DataProcessor::process_data(items);
    }
};

struct SimpleData {
    int id;
    std::string name;
    
    SimpleData(int id, const std::string& name) : id(id), name(name) {}
};

// Top-level functions
int main() {
    auto processor = DataProcessor::create_default();
    std::vector<int> data = {1, 2, 3, 4, 5};
    
    processor->process_data(data);
    
    const auto& result = processor->get_data();
    for (const auto& item : result) {
        std::cout << item << " ";
    }
    std::cout << std::endl;
    
    return 0;
}